File: app/src/main/java/io/goooler/demoapp/activity/MainActivity.java
package io.goooler.demoapp.activity;

import android.os.Bundle;
import android.view.View;

import androidx.databinding.DataBindingUtil;

import io.goooler.demoapp.R;
import io.goooler.demoapp.async.ClickHandler;
import io.goooler.demoapp.base.BaseActivity;
import io.goooler.demoapp.base.BaseApplication;
import io.goooler.demoapp.databinding.ActivityMainBinding;
import io.goooler.demoapp.util.ToastUtil;

public class MainActivity extends BaseActivity implements ClickHandler {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        binding.setClickHandler(this);
    }

    @Override
    protected void onDestroy() {
        BaseApplication.destroyGlobalObject();
        super.onDestroy();
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.buttonPanel) {
            ToastUtil.showToast("button clicked");
        }
    }
}

File: app/src/main/java/io/goooler/demoapp/activity/MainActivity.kt
package io.goooler.demoapp.activity

import android.net.Uri
import android.os.Bundle
import android.view.View

import androidx.databinding.DataBindingUtil

import io.goooler.demoapp.R
import io.goooler.demoapp.async.ClickHandler
import io.goooler.demoapp.base.BaseActivity
import io.goooler.demoapp.base.BaseApplication
import io.goooler.demoapp.databinding.ActivityMainBinding
import io.goooler.demoapp.util.ToastUtil

class MainActivity : BaseActivity(), BlankFragment.OnFragmentInteractionListener {

    private val binding by lazy {
        DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding
        addFragment(R.id.fragment_root, BlankFragment.newInstance())
    }

    override fun onFragmentInteraction(uri: Uri) {
        //do nothing
    }

    override fun onDestroy() {
        BaseApplication.destroyGlobalObject()
        super.onDestroy()
    }
}
